src/components/PostEditor/PostEditor.js

```diff
@@ -327,26 +328,28 @@ export default class PostEditor extends React.Component {
     )
   }
 
-  save = () => {
+  save = async () => {
     const {
       editing, createPost, updatePost, onClose,
       goToPost, setAnnouncement, announcementSelected,
-      imageAttachments, fileAttachments
+      imageAttachments, fileAttachments, fetchLocation, ensureLocationIdIfCoordinate
     } = this.props
     const {
       id, type, title, groups, linkPreview, members,
       acceptContributions, eventInvitations, startTime,
       endTime, location, locationId, isPublic
     } = this.state.post
+    let coordLocationId = null
     const details = this.editor.current.getContentHTML()
     const topicNames = this.topicSelector.current.getSelected().map(t => t.name)
     const memberIds = members && members.map(m => m.id)
     const eventInviteeIds = eventInvitations && eventInvitations.map(m => m.id)
     const imageUrls = imageAttachments && imageAttachments.map(attachment => attachment.url)
     const fileUrls = fileAttachments && fileAttachments.map(attachment => attachment.url)
+    coordLocationId = await ensureLocationIdIfCoordinate({ fetchLocation, location, locationId })
```

Comment:
this feels smelly, shouldnt all the location stuff happen in the polling so that by the time you save its all figured out?

Reply:
`pollingFetchLocation` is only triggered if you select an option from the autocomplete drop-down. Coordinates pasted or typed in will not provide a result to select.  So coordinates need to be handled in another way.

If we offer the fullText as a selectable option in the autocomplete list, in the coordinate case, the user might not select the coordinate fullText anyway. And in the non-coordinate case, we might give the user the impression that an address is correct (and shows up in the geocoder) when it doesn't actually exist.